contrib/src/main/java/com/datatorrent/contrib/geode/GeodeStore.java

```diff
@@ -0,0 +1,308 @@
+   *          the clientCache to set
+   */
+  public void setClientCache(ClientCache clientCache)
+  {
+    this.clientCache = clientCache;
+  }
+
+  /**
+   * @return the locatorPort
+   */
+  public int getLocatorPort()
+  {
+    return locatorPort;
+  }
+
+  /**
+   * @param locatorPort
+   *          the locatorPort to set
+   */
+  public void setLocatorPort(int locatorPort)
+  {
+    this.locatorPort = locatorPort;
+  }
+
+  /**
+   * @return the locatorHost
+   */
+  public String getLocatorHost()
+  {
+    return locatorHost;
+  }
+
+  /**
+   * @param locatorHost
+   *          the locatorHost to set
+   */
+  public void setLocatorHost(String locatorHost)
+  {
+    this.locatorHost = locatorHost;
+  }
+
+  /**
+   * @return the region
+   * @throws IOException
+   */
+  public Region<Object, Object> getRegion() throws IOException
+  {
+    // return region;
+    if (clientCache == null && clientCache.isClosed()) {
+      initClient();
+    }
+
+    if (region == null) {
+      region = clientCache.getRegion(regionName);
+      if (region == null) {
+        region = clientCache.<Object, Object> createClientRegionFactory(ClientRegionShortcut.PROXY).create(regionName);
+      }
+    }
+
+    return region;
+  }
+
+  /**
+   * @param region
+   *          the region to set
+   */
+  public void setRegion(Region<Object, Object> region) throws IOException
+  {
+    this.region = region;
+  }
+
+  @Override
+  public void connect() throws IOException
+  {
+    try {
+      clientCache = new ClientCacheFactory().addPoolLocator(getLocatorHost(), getLocatorPort()).create();
+    } catch (CacheClosedException ex) {
+      LOG.info("error initiating client ", ex);
+    }
+
+    region = clientCache.getRegion(getRegionName());
+
+    if (region == null) {
+      region = clientCache.<Object, Object> createClientRegionFactory(ClientRegionShortcut.PROXY).create(
+          getRegionName());
+    }
+
+  }
+
+  @Override
+  public void disconnect() throws IOException
+  {
+    clientCache.close();
+
+  }
+
+  @Override
+  public boolean isConnected()
+  {
+    return (clientCache.isClosed());
+
+  }
+
+  /**
+   * Gets the value given the key. Note that it does NOT work with hash values
+   * or list values
+   *
+   * @param key
+   * @return The value.
+   */
+  @Override
+  public Object get(Object key)
+  {
+
+    try {
+      return (getRegion().get(key));
+    } catch (IOException ex) {
+      LOG.info("error getting object ", ex);
+      return null;
+    }
+
+  }
+
+  /**
+   * Gets all the values given the keys. Note that it does NOT work with hash
+   * values or list values
+   *
+   * @param keys
+   * @return All values for the given keys.
+   */
+  @SuppressWarnings("unchecked")
+  @Override
+  public List<Object> getAll(List<Object> keys)
+  {
+
+    List<Object> values = new ArrayList<Object>();
+
+    try {
+      final Map<Object, Object> entries = getRegion().getAll(keys);
+      for (int i = 0; i < keys.size(); i++) {
```

Comment:
The method expects the values to be returned in the same order as the keys are passed. However apache geode returns a map of key values. Hence to maintain the semantics have iterated the entries and returned just the values in the order of the keys passed.